### DISCUSSION TITLE
User defined config? (And merge efforts?)

### DISCUSSION THREAD

**Discussion by Zykino** [topgrade-rs/topgrade#363] [Show and tell]
I recently started doing [a project](https://zykino.net/gitea/zykino/system-updater) and even more recently heard about `topgrade`.

Turns out both projects have a similar objective: update user’s system. The approach is quite different because I wanted the users to be able to configure any program useful to them. For example here is my setup to [update `helix`](https://zykino.net/gitea/zykino/system-updater/src/branch/master/examples/git-helix.yaml) installed with `git` & `cargo install --path`.

My naming and format are still evolving… but I think what I call "packages" is `topgrade`’s executors. Ask me if you want clarifications on any-thing.

What I have come up with is having any number of command as `fetch`, then `compile` commands to execute in preparation. The `install` one is the only required and is unique. And then, another list of commands for post-installation actions.
I took inspiration on `apt`’s steps, and usual git installations setups too.

Maybe having 2 districts projects in the same language to achieve the same objective is too much. Do you think it would be possible to integrate some kind of config like my yamls?

**Comment by s34m** [comment]
This would definitely be something to think about, there are definitely a lot of things we could merge together. I especially like the system of user configured commands you did, which is better then our system.
